src/observer.rs

```diff
@@ -424,8 +419,8 @@ mod test {
     async fn request_handler(req: CoapRequest<SocketAddr>) -> Option<CoapResponse> {
         match req.get_method() {
             &coap_lite::RequestType::Get => {
-                let observe_option = req.message.get_observe().unwrap();
-                assert_eq!(observe_option[0], ObserveOption::Deregister as u8);
+                let observe_option = req.message.get_observe_value().unwrap();
```

Comment:
`observe_option` is wrong type. The line should be `let observe_option = req.get_observe_flag().unwrap().unwrap();`